### PR TITLE
Add macOS build to travis.

### PR DESCRIPTION
**Pull request by simonnagl**
Fixes macOS part of https://github.com/firehol/netdata/issues/1667

### CODE REVIEW

**Comment by simonnagl** [comment]
This is mergeable from my point of view.

@glensc, @philwhineray could one of you please review?

File: .travis.yml [hunk 1]
```diff
@@ -47,5 +52,6 @@ deploy:
     file: "netdata*.tar.*"
     on:
       condition: $CC = gcc
+      condition: $TRAVIS_OS_NAME = linux
```
Comment (philwhineray) [inline .travis.yml]: According to [this](https://docs.travis-ci.com/user/deployment/) you can only have one condition.

Comment (philwhineray) [inline .travis.yml]: Otherwise seems OK to me.

Comment (simonnagl) [inline .travis.yml]: Ok, then removing this line would be appropriate?

Comment (philwhineray) [inline .travis.yml]: I guess it wants merging with the line above e.g. with &&, otherwise multiple builds will try to deploy their source files.

Comment (simonnagl) [inline .travis.yml]:
```
if [[ $CC = gcc && $TRAVIS_OS_NAME = linux ]]; then echo test; fi
```
works on the command line. I'll fix condition. Thank you for finding the error now!

Comment (simonnagl) [inline .travis.yml]: Done.

**Comment by philwhineray** [comment]
Sorry for the delay. Was out and about.

Looks good to me now and glad to be of assistance.

**Comment by ktsaou** [comment]
so, merge?

**Comment by philwhineray** [comment]
Yes.

**Comment by simonnagl** [comment]
Yes

**Comment by ktsaou** [comment]
merged! thanks!

File: .travis.yml [hunk 2]
```diff
@@ -46,6 +51,6 @@ deploy:
     file_glob: true
     file: "netdata*.tar.*"
     on:
-      condition: $CC = gcc
+      condition: $CC = gcc && $TRAVIS_OS_NAME
```
Comment (glensc) [inline .travis.yml]: `$TRAVIS_OS_NAME` is always true isn't it? you mean `$TRAVIS_OS_NAME == "linux"` perhaps?

Comment (ktsaou) [inline .travis.yml]: hm.. I think this comment is for @philwhineray .
@philwhineray could you please check it?

**Comment by philwhineray** [comment]
Yes, I think it should have = linux specified (assuming that is what gets
set in the VM).

It is probably not fatal as-is because the tar-files will just be uploaded
multiple times but it would be better if only one build is responsible for
doing so.

File: .travis.yml (hunk 2 above)
Comment (simonnagl) [inline .travis.yml]: @glensc Yes, this is correct. Sorry for the mistake. @ktsaou I can make a pull request but it seems more easy if you fix this in one of your next "minor fix PR". Is that ok for you?

Comment (ktsaou) [inline .travis.yml]: thanks! it is already fixed: https://github.com/ktsaou/netdata/commit/2f756fa58aba243cd6579270ff93de28c09db1e5

**Comment by glensc** [comment]
@philwhineray `overwrite: true` is not set in `.travis.yml`, so only first build will upload and others silently skip.

**Comment by philwhineray** [comment]
If you are referring to my commit message, not true: the overwrite happens
on branch builds using a custom script. This is distinct from the github
releases deployment, which happens on tagged versions.

**Pull request merged**